### PR TITLE
Graduate Quota configuration API to v1beta1

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
ref: kubernetes/features#587

**Release note**:

```release-note
Quota admission configuration api graduated to v1beta1
```
/sig node
/sig scheduling
/cc @derekwaynecarr @deads2k @liggitt @sjenning @aveshagarwal @ravisantoshgudimetla @smarterclayton

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
@vikaschoudhary16 - Can you please look at the lint failures:

Do you want to modify e2e's to use beta version in a follow-up PR?

**Comment by vikaschoudhary16** [comment]
@ravisantoshgudimetla golint should be happy now.

> Do you want to modify e2e's to use beta version in a follow-up PR?

There is no existing e2e test which is using this api. Also quoting @aveshagarwal
https://github.com/kubernetes/kubernetes/pull/64645#issuecomment-394350428
> varying admission plugin config can not be passed to e2e tests, as that would require restarting the cluster whenever the config is changed.

Though i am still thinking how to add an e2e for this. Will create a different PR, if i figured it out. Meanwhile would like to know suggestions if got any.

**Comment by k82cn** [comment]
LGTM overall :)

**Comment by ravisantoshgudimetla** [comment]
@vikaschoudhary16 The way I have done moving api from alpha to beta in the past is ensuring alpha versionexists while creating beta. This ensures people using alpha have some breathing space before switching to beta. Please take a look at https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md, while the wording is not clear on removing alpha version, it explicitly tells to copy https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#making-a-new-api-version

File: plugin/pkg/admission/resourcequota/apis/resourcequota/v1beta1/BUILD [hunk 1]
```diff
@@ -16,7 +16,7 @@ go_library(
         "zz_generated.deepcopy.go",
         "zz_generated.defaults.go",
     ],
-    importpath = "k8s.io/kubernetes/plugin/pkg/admission/resourcequota/apis/resourcequota/v1alpha1",
```
Comment (bsalamat) [inline plugin/pkg/admission/resourcequota/apis/resourcequota/v1beta1/BUILD]: Please do not delete (rename) v1alpha1 files. If we deleted these files, v1alpha1 APIs would stop working in the next K8s release.

File: plugin/pkg/admission/resourcequota/BUILD [hunk 2]
```diff
@@ -28,7 +28,7 @@ go_library(
         "//pkg/util/workqueue/prometheus:go_default_library",
         "//plugin/pkg/admission/resourcequota/apis/resourcequota:go_default_library",
         "//plugin/pkg/admission/resourcequota/apis/resourcequota/install:go_default_library",
-        "//plugin/pkg/admission/resourcequota/apis/resourcequota/v1alpha1:go_default_library",
```
Comment (ravisantoshgudimetla) [inline plugin/pkg/admission/resourcequota/BUILD]: Shouldn't we have this line?

File: plugin/pkg/admission/resourcequota/apis/resourcequota/install/install.go [hunk 3]
```diff
@@ -22,12 +22,12 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	resourcequotaapi "k8s.io/kubernetes/plugin/pkg/admission/resourcequota/apis/resourcequota"
-	resourcequotav1alpha1 "k8s.io/kubernetes/plugin/pkg/admission/resourcequota/apis/resourcequota/v1alpha1"
```
Comment (ravisantoshgudimetla) [inline plugin/pkg/admission/resourcequota/apis/resourcequota/install/install.go]: I think, we should have this line as well.

File: plugin/pkg/admission/resourcequota/apis/resourcequota/install/install.go [hunk 4]
```diff
@@ -22,12 +22,12 @@ import (
...
+	resourcequotav1beta1 "k8s.io/kubernetes/plugin/pkg/admission/resourcequota/apis/resourcequota/v1beta1"
 )
 
 // Install registers the API group and adds types to a scheme
 func Install(scheme *runtime.Scheme) {
 	utilruntime.Must(resourcequotaapi.AddToScheme(scheme))
-	utilruntime.Must(resourcequotav1alpha1.AddToScheme(scheme))
```
Comment (ravisantoshgudimetla) [inline plugin/pkg/admission/resourcequota/apis/resourcequota/install/install.go]: I think you should have this version but setting the default version to beta in subsequent line which you are doing should be good enough.

Comment (vikaschoudhary16) [reply plugin/pkg/admission/resourcequota/apis/resourcequota/install/install.go]: actually both should be added in subsequent line in priority order. will update, Thanks!

**Review by bsalamat — APPROVED** [review]
/lgtm

Thanks, @vikaschoudhary16!

**Pull request merged**